### PR TITLE
Module/compressed items

### PR DESCRIPTION
**Pull request by YTG1234**
Adds compressed blocks for items and stuff :)

### CODE REVIEW

File: src/main/java/io/github/hydos/proton/module/building/CompressedItemsModule.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+            BLUER_ICE = ProtonRegisterUtil.block("bluer_ice",
+                                                 new Block(FabricBlockSettings.copyOf(Blocks.BLUE_ICE)
+                                                                              .slipperiness(0.9998F)),
+                                                 new FabricItemSettings().group(ItemGroup.BUILDING_BLOCKS)
+                                                );
+
+            DARK_BLUE_ICE = ProtonRegisterUtil.block("dark_blue_ice",
+                                                 new Block(FabricBlockSettings.copyOf(Blocks.BLUE_ICE)
+                                                                              .slipperiness(1.234F)),
```
Comment (YTG1234) [inline src/main/java/io/github/hydos/proton/module/building/CompressedItemsModule.java]: @dzwdz Forced me to do this

File: src/main/resources/proton.accessWidener [hunk 2]
```diff
@@ -0,0 +1,2 @@
+accessWidener   v1  named
+accessible  method net/minecraft/block/Blocks   never   (Lnet/minecraft/block/BlockState;Lnet/minecraft/world/BlockView;Lnet/minecraft/util/math/BlockPos;)Z
```
Comment (YTG1234) [inline src/main/resources/proton.accessWidener:2]: I could've just created a lambda but why not

Comment (dzwdz) [reply src/main/resources/proton.accessWidener:2]: no

File: src/main/java/io/github/hydos/proton/module/building/CompressedItemsModule.java (hunk 1 above)
Comment (dzwdz) [inline src/main/java/io/github/hydos/proton/module/building/CompressedItemsModule.java]: just set it to 1

**Review by dzwdz — CHANGES_REQUESTED** [review]
ok the bluest ice thing was funny but we can't just break the game
also please get rid of the access widener

File: build.gradle [hunk 3]
```diff
@@ -23,6 +23,10 @@ shadowJar {
 	relocate 'nonapi.io.github.classgraph', 'proton.shaded.nonapi.io.github.classgraph'
 }
 
+minecraft {
+	accessWidener "src/main/resources/proton.accessWidener"
```
Comment (hYdos) [inline build.gradle:27]: access widener best. accessor bad

File: src/main/java/io/github/hydos/proton/module/building/CompressedItemsModule.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public static class ModuleBlocks {
+        public static Block COMPRESSED_NETHER_STAR;
+        public static Block BLUER_ICE;
+        public static Block DARK_BLUE_ICE;
+
+        public static void register() {
+            COMPRESSED_NETHER_STAR = ProtonRegisterUtil.block("compressed_nether_star",
+                                                              new Block(FabricBlockSettings.of(Material.STONE)
```
Comment (hYdos) [inline src/main/java/io/github/hydos/proton/module/building/CompressedItemsModule.java]: the formatting here kills me. if ur in IDEA, please ctrl + shift + l

File: src/main/resources/proton.accessWidener (hunk 2 above)
Comment (hYdos) [reply src/main/resources/proton.accessWidener:2]: accesswiderner good.

**Review by hYdos — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/hydos/proton/module/building/CompressedItemsModule.java (hunk 4 above)
Comment (YTG1234) [reply src/main/java/io/github/hydos/proton/module/building/CompressedItemsModule.java]: Maybe I didn't set up my code format settings correctly 🥇

File: src/main/java/io/github/hydos/proton/module/building/CompressedItemsModule.java (hunk 1 above)
Comment (YTG1234) [reply src/main/java/io/github/hydos/proton/module/building/CompressedItemsModule.java]: Done

**Review by hYdos — APPROVED** [review]

**Comment by YTG1234** [comment]
Bonus: This PR also makes the mod menu config button open the config screen

File: src/main/java/io/github/hydos/proton/mixin/building/compressed_items/ItemEntityMixin.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(ItemEntity.class)
+public abstract class ItemEntityMixin {
+    @Shadow public abstract ItemStack getStack();
+
+    @Inject(method = "damage(Lnet/minecraft/entity/damage/DamageSource;F)Z", at = @At("RETURN"), cancellable = true)
+    private void noDestroyCompressedNetherStars(DamageSource source, float amount, CallbackInfoReturnable<Boolean> cir) {
+        if (!getStack().isEmpty() && getStack().getItem() == CompressedItemsModule.ModuleBlocks.COMPRESSED_NETHER_STAR.asItem() && source.isExplosive()) {
```
Comment (dzwdz) [inline src/main/java/io/github/hydos/proton/mixin/building/compressed_items/ItemEntityMixin.java]: just make it use tags

Comment (YTG1234) [reply src/main/java/io/github/hydos/proton/mixin/building/compressed_items/ItemEntityMixin.java]: Are there any other items?

Comment (YTG1234) [reply src/main/java/io/github/hydos/proton/mixin/building/compressed_items/ItemEntityMixin.java]: something like `proton:survives_explosion`?

**Review by dzwdz — APPROVED** [review]
i don't see any obvious problems

**Pull request merged**